src/functions_framework/__init__.py

```diff
@@ -68,18 +72,49 @@ def view_func(path):
     return view_func
 
 
-def _is_binary_cloud_event(request):
-    return (
-        request.headers.get("ce-type")
-        and request.headers.get("ce-specversion")
-        and request.headers.get("ce-source")
-        and request.headers.get("ce-id")
-    )
+def _run_cloudevent(function, request):
+    data = request.get_data()
+    event = from_http(data, request.headers)
+    function(event)
+
+
+def _cloudevent_view_func_wrapper(function, request):
+    def view_func(path):
+        try:
+            _run_cloudevent(function, request)
+        except (
+            cloud_exceptions.CloudEventMissingRequiredFields,
+            cloud_exceptions.CloudEventTypeErrorRequiredFields,
+        ) as e:
+            flask.abort(
+                400,
+                description=(
+                    "Function was defined with FUNCTION_SIGNATURE_TYPE=cloudevent but"
+                    " failed to find all required cloudevent fields. Found http"
+                    f" headers: {request.headers} and data: {request.get_data()}. "
+                    f"cloudevents.exceptions.{type(e).__name__}: {e}"
+                ),
+            )
+        except json.decoder.JSONDecodeError as e:
+            # TODO: more detailed error messages or error handling in sdk-python
+            flask.abort(
+                400,
+                description=(
+                    f"{e} "
```

Comment:
```suggestion
```
I don't think we want the exception message as the first line of the error here. It's already included at the end as well.